chromadb/telemetry/events.py

```diff
@@ -13,11 +13,76 @@ class ServerStartEvent(TelemetryEvent):
     name: ClassVar[str] = "server_start"
 
 
+@dataclass
+class ClientCreateCollectionEvent(TelemetryEvent):
+    name: ClassVar[str] = "client_create_collection"
+    collection_uuid: str
+    embedding_function: str
+
+
 @dataclass
 class CollectionAddEvent(TelemetryEvent):
     name: ClassVar[str] = "collection_add"
     collection_uuid: str
     add_amount: int
+    with_embeddings: bool
+    with_metadata: bool
+    with_documents: bool
+    batch_size = 4
+
+    def can_batch(self, other: TelemetryEvent) -> bool:
+        return (
+            isinstance(other, CollectionAddEvent)
+            and self.collection_uuid == other.collection_uuid
+            and self.with_embeddings == other.with_embeddings
+            and self.with_metadata == other.with_metadata
+            and self.with_documents == other.with_documents
+        )
+
+    def batch(self, other: "TelemetryEvent") -> "CollectionAddEvent":
+        if not self.can_batch(other):
+            raise ValueError("Cannot batch events")
+        other = cast(CollectionAddEvent, other)
+        return CollectionAddEvent(
+            collection_uuid=self.collection_uuid,
+            add_amount=self.add_amount + other.add_amount,
```

Comment:
One thing we lose with this, is the amount added per call to add() - which is useful to know the usage pattern around. We could either:

1. Add an average
2. Add an array
3. Drop it and assume that current historical usage is sufficient.

Reply:
Do we ship the batch size?

Reply:
We should!